api_routes.js

```diff
@@ -1,29 +1,19 @@
 'use strict';
 const { extension } = require('./extension');
 const express = require('express');
-const { sessionMiddleware } = require('./middleware/session_middleware');
-const { ApplicationConfig, ApplicationClient } = require("@gofynd/fdk-client-javascript");
-
+const { sessionMiddleware, getApplicationConfig } = require('./middleware/session_middleware');
 
 function setupProxyRoutes() {
     const apiRoutes = express.Router({  mergeParams: true });
     const applicationProxyRoutes = express.Router({  mergeParams: true });
 
     applicationProxyRoutes.use(async (req, res, next) => {
         try {
-            if(req.headers["x-user-data"]) {
-                req.user = JSON.parse(req.headers["x-user-data"]);
-                req.user.user_id = req.user._id;
-            }
-            if(req.headers["x-application-data"]) {
-                req.application = JSON.parse(req.headers["x-application-data"]);
-                req.applicationConfig = new ApplicationConfig({
-                    applicationID: req.application._id,
-                    applicationToken: req.application.token,
-                    domain: extension.cluster
-                });
-                req.applicationClient = new ApplicationClient(req.applicationConfig);
-            }
+            const { user, application, applicationConfig, applicationClient } = await getApplicationConfig(req.headers["x-user-data"], req.headers["x-application-data"], extension)
```

Comment:
Check if we can define two different methods for application and user @jinalvirani

Reply:
Done